android-core/src/main/java/com/mparticle/internal/HashingUtility.kt

```diff
@@ -0,0 +1,66 @@
+package com.mparticle.internal
+
+import com.mparticle.MPEvent
+import com.mparticle.MParticle
+import com.mparticle.internal.messages.BaseMPMessage
+
+/**
+ * Simple bit-shifting hash for use with filtering. mParticle's backend uses this same hash
+ * to compute hashes of events and their attributes, and will include those hashes with
+ * configuration settings for kits.
+ *
+ * @return return int hash
+ */
+object HashingUtility {
+
+    fun hashEvent(event: MPEvent): Int =
+        MPUtility.mpHash("${event.eventType.ordinal}${event.eventName}")
+
+    fun hashMpMessageTypeName(mpMessage: BaseMPMessage): Int =
+        MPUtility.mpHash("${mpMessage.messageType}${mpMessage.name}")
+
+    fun hashFilterAvKey(key: String): Int = MPUtility.mpHash(key)
```

Comment:
Why split these up into two (hashFilterAvKey & hashFilterAvValue) separate methods? Seems like an unnecessary complexity.

Reply:
Ben asked me to create functions in the utility class based on semantic so because conceptually there could be different ways to hash those (since conceptually those are different things) we wanted to have the separately event though the implementation is the same.

Reply:
Yes that's correct, it's some slight code duplication but makes the purpose clearer at the call site. Plus if we ever change the hash method in the future (unlikely but possible) it's more flexible, but the main reason is the former.